File: domain/src/main/kotlin/com/readler/domain/repository/AiQueuedRequest.kt
package com.readler.domain.repository

import java.util.UUID

enum class AiRequestType {
    QUESTION,
    EXPLAIN_SELECTION,
    TRANSLATE_SELECTION,
    SIMILAR_BOOKS,
    SECTION_SUMMARY
}

data class AiQueuedRequest(
    val requestId: String = UUID.randomUUID().toString(),
    val bookId: String,
    val bookTitle: String,
    val author: String?,
    val type: AiRequestType,
    val prompt: String,
    val selectedText: String? = null,
    val targetLanguage: String? = null,
    val sectionContext: String? = null,
    val createdAtEpochMs: Long = System.currentTimeMillis()
)

File: lib/domain/repository/ai_queued_request.dart
import 'package:uuid/uuid.dart';

enum AiRequestType {
  question,
  explainSelection,
  translateSelection,
  similarBooks,
  sectionSummary,
}

class AiQueuedRequest {
  final String requestId;
  final String bookId;
  final String bookTitle;
  final String? author;
  final AiRequestType type;
  final String prompt;
  final String? selectedText;
  final String? targetLanguage;
  final String? sectionContext;
  final int createdAtEpochMs;

  AiQueuedRequest({
    String? requestId,
    required this.bookId,
    required this.bookTitle,
    this.author,
    required this.type,
    required this.prompt,
    this.selectedText,
    this.targetLanguage,
    this.sectionContext,
    int? createdAtEpochMs,
  })  : requestId = requestId ?? const Uuid().v4(),
        createdAtEpochMs =
            createdAtEpochMs ?? DateTime.now().millisecondsSinceEpoch;
}
